platform/web/audio_driver_web.cpp

```diff
@@ -105,7 +106,10 @@ void AudioDriverWeb::_audio_driver_capture(int p_from, int p_samples) {
 Error AudioDriverWeb::init() {
 	int latency = Engine::get_singleton()->get_audio_output_latency();
 	if (!audio_context.inited) {
-		audio_context.mix_rate = _get_configured_mix_rate();
+		int mix_rate = _get_configured_mix_rate();
+		audio_context.mix_rate = mix_rate == 0
+				? godot_audio_get_browser_mix_rate()
+				: mix_rate;
```

Comment:
This change is not really needed because `audio_context.mix_rate` is passed as a pointer to `godot_audio_init` which sets it to the actual resulting `AudioContext.sampleRate`.

```suggestion
		// If 0, it will be set to the browser preferred value after calling godot_audio_init.
		audio_context.mix_rate = _get_configured_mix_rate();
```

While it might be tempting to have a `godot_audio_get_browser_mix_rate()` function, this forces us to create a new AudioContext which:
- Cause an extra console warning if audiocontext was not allowed to start (not a result of a user action)
- Uses a part of the spec (opening multiple `AudioContext`s) which is [under-standardized](https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/AudioContext#usage_notes) (especially given we can't await for the first one to close).

In practice, this would likely be "safe enough", but I've learned not to trust vendors in maintaining compatibility with poorly standardized behaviors.

So, given we don't seem to need that function anywhere else, I suggest to remove it completely.

Reply:
@Faless I added this recently because it causes issues on Safari otherwise (the other browsers are fine).

Reply:
I'll find a way to only use one `AudioContext`.

Reply:
> I added this recently because it causes issues on Safari otherwise (the other browsers are fine).

In theory this change should not affect browser behavior.

With the change in the PR:
- We create an audio context without specifying any parameter.
- We read the assigned mix rate.
- Create a the true audio context with this "default" mix rate.
- Return the resulting audio context mix rate (sampleRate) to C++ as an out parameter.

With my change (and as currently in upstream):
- We create an audio context specifying the mix rate *only if* it's not zero.
- Return the resulting audio context mix rate (sampleRate) to C++ as an out parameter.

If this change affects Safari, it might be an issue with the way we set/read the sample rate. Try printing the values on both C++ `audio_context.mix_rate` and JS side `ctx.sampleRate`.

If all of the above seems to be correctly set (and it's not a bug in some of my old code :sweat_smile: ), we might be hitting different internal code path in Safari (when the sampleRate is and is not specified). Browsers usually greatly prefer not being forced a specific mix rate, but Safari has always been the weird of the bunch, and I currently can't test that specifically.